### ISSUE TITLE
Restart doesn't happen after change-detection

### ISSUE THREAD

**Issue by flash1293** [remy/nodemon#626]
I have a nodemon running in a docker-container with a mounted volume on OSX. Nodemon is receiving the file-change but it doesn't restart the server (somewhere between detecting the file-change and actually restarting something seems to go wrong).

Output:

    Creating piclet_web_1...
    Attaching to piclet_web_1
    web_1 | 7 Sep 20:08:06 - [nodemon] v1.4.1
    web_1 | 7 Sep 20:08:06 - [nodemon] to restart at any time, enter `rs`
    web_1 | 7 Sep 20:08:06 - [nodemon] ignoring: .git /app/node_modules/**/* bower_components .sass-cache
    web_1 | 7 Sep 20:08:06 - [nodemon] watching: *.*
    web_1 | 7 Sep 20:08:06 - [nodemon] watching extensions: js
    web_1 | 7 Sep 20:08:06 - [nodemon] starting `node index.js`
    web_1 | 7 Sep 20:08:06 - [nodemon] child pid: 20
    web_1 | 7 Sep 20:08:06 - [nodemon] watching 8 files
    web_1 | restify listening at http://[::]:80 //normal output of started node-app
    web_1 | 7 Sep 20:08:12 - [nodemon] files triggering change check: lib/server.js //file-change, which gets detected
    web_1 | 7 Sep 20:08:12 - [nodemon] matched rule: **/*.*
    web_1 | 7 Sep 20:08:12 - [nodemon] changes after filters (before/after): 1/1
    web_1 | 7 Sep 20:08:12 - [nodemon] restarting due to changes...
    web_1 | 7 Sep 20:08:12 - [nodemon] lib/server.js
    web_1 | 
    //no output of restarted server

Dockerfile:

    FROM node:0.12.7-wheezy
    EXPOSE 80
    
    RUN npm install nodemon -g
    
    ADD app /app
    WORKDIR app
    
    RUN npm install

docker-compose.yml

    web:
      build: .
      ports:
        - "80:80"
      volumes:
        - ./app/lib:/app/lib
      links:
        - db
      command: "nodemon -L ./index.js"
    db:
      image: mongo:3.1.7

--dump output:
```
    web_1 | node: v0.12.7
    web_1 | nodemon: v1.4.1
    web_1 | command: node /usr/local/bin/nodemon -L -V --dump
    web_1 | cwd: /app
    web_1 | OS: linux x64
    web_1 | --------------
    web_1 | { run: false,
    web_1 |   system:
    web_1 |    { cwd: '/app',
    web_1 |      useFind: true,
    web_1 |      useWatch: false,
    web_1 |      useWatchFile: false },
    web_1 |   required: false,
    web_1 |   dirs: [ '/app' ],
    web_1 |   timeout: 1000,
    web_1 |   options:
    web_1 |    { legacyWatch: true,
    web_1 |      verbose: true,
    web_1 |      dump: true,
    web_1 |      ignore:
    web_1 |       [ '.git',
    web_1 |         'node_modules',
    web_1 |         'bower_components',
    web_1 |         '.sass-cache',
    web_1 |         re: /\.git|node_modules|bower_components|\.sass\-cache/ ],
    web_1 |      watch: [ '*.*', re: /.*\..*/ ],
    web_1 |      restartable: 'rs',
    web_1 |      execMap: { py: 'python', rb: 'ruby' },
    web_1 |      stdin: true,
    web_1 |      runOnChangeOnly: false,
    web_1 |      stdout: true,
    web_1 |      execOptions:
    web_1 |       { script: 'index.js',
    web_1 |         exec: 'node',
    web_1 |         args: [],
    web_1 |         scriptPosition: 0,
    web_1 |         nodeArgs: undefined,
    web_1 |         ext: 'js',
    web_1 |         env: {},
    web_1 |         execArgs: [] },
    web_1 |      monitor:
    web_1 |       [ '*.*',
    web_1 |         '!.git',
    web_1 |         '!/app/node_modules/**/*',
    web_1 |         '!bower_components',
    web_1 |         '!.sass-cache' ] },
    web_1 |   load: [Function],
    web_1 |   reset: [Function: reset],
    web_1 |   lastStarted: 0,
    web_1 |   loaded: [],
    web_1 |   command:
    web_1 |    { raw: { executable: 'node', args: [ 'index.js' ] },
    web_1 |      string: 'node index.js' },
    web_1 |   offset: 0 }
    web_1 | --------------
```

**Comment by flash1293** [comment]
For the record, I managed to get it to work by changing the base-image of the container from `node:0.12.7-wheezy` to `node:0.12.7`. But I think it's still a very strange behavior which should get investigated.

**Comment by remy** [comment]
Please retest with new development build - full details in this thread #625

**Comment by flash1293** [comment]
Still the same problem...

**Comment by remy** [comment]
Can you test and confirm the container can run and send `kill -SIGUSR2`

This is the likely cause.

**Comment by flash1293** [comment]
Yes, `kill -SIGUSR2` works. It's and error-handling problem - nodemon should somehow communicate that the restart failed and not just silently fail.

**Comment by remy** [comment]
Can you debug the container then?

**Comment by flash1293** [comment]
I'm not exactly sure what you want me to do, sorry...

**Comment by remy** [comment]
Well...you have a bespoke environment that you're hitting this problem in. You're using the latest 1.5-alpha, right? So I need you to try to dig out the actual line in nodemon that's causing the issue and help with the resolution, otherwise I'm blind on how to approach the problem.

**Comment by remy** [comment]
#625 closes and released in `nodemon@1.4.2`
